x/deployment/types/group_pricing_validation.go

```diff
@@ -0,0 +1,78 @@
+package types
+
+import (
+	"fmt"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ovrclk/akash/validation/constants"
+	"github.com/pkg/errors"
+
+	"github.com/ovrclk/akash/types/unit"
+)
+
+func validateGroupPricing(gspec GroupSpec) error {
+	var price sdk.Coin
+
+	mem := sdk.NewInt(0)
+
+	for idx, resource := range gspec.Resources {
+
+		if err := validateUnitPricing(resource); err != nil {
+			return fmt.Errorf("group %v: %w", gspec.GetName(), err)
+		}
+
+		if resource.FullPrice().Denom != constants.AkashDenom {
+			return fmt.Errorf("%w: denomination must be %q", ErrInvalidDeployment, constants.AkashDenom)
+		}
+
+		if idx == 0 {
+			price = resource.FullPrice()
+		} else {
+			rprice := resource.FullPrice()
+			if rprice.Denom != price.Denom {
+				return errors.Errorf("multi-denonimation group: (%v == %v fails)", rprice.Denom, price.Denom)
+			}
+			price = price.Add(rprice)
+		}
+
+		memCount := sdk.NewInt(0)
+		if u := resource.Resources.Memory; u != nil {
+			memCount.Add(sdk.NewIntFromUint64(u.Quantity.Value()))
+		}
+
+		mem = mem.Add(memCount.Mul(sdk.NewIntFromUint64(uint64(resource.Count))))
+	}
+
+	minprice := mem.Mul(sdk.NewInt(validationConfig.MinGroupMemPrice)).Quo(sdk.NewInt(unit.Gi))
+
+	if price.Amount.LT(minprice) {
+		return errors.Errorf("group %v: price too low (%v >= %v fails)", gspec.GetName(), price, minprice)
+	}
+
+	// TODO - validate uakt for everything
+	return nil
+}
+
+func validateUnitPricing(rg Resource) error {
+	if !rg.GetPrice().IsValid() {
+		return errors.Errorf("error: invalid price object")
+	}
+
+	if rg.Price.Amount.GT(sdk.NewIntFromUint64(uint64(validationConfig.MaxUnitPrice))) {
+		return errors.Errorf("error: invalid unit price (%v > %v fails)", validationConfig.MaxUnitPrice, rg.Price)
+	}
+
+	if rg.Price.Amount.GT(sdk.NewIntFromUint64(uint64(validationConfig.MaxUnitPrice))) {
```

Comment:
I think this should be *L*ess*T*han() based on the error message, and that it's the same comparison above?

Reply:
I think I just re-enabled this code, I'll double check it.

Reply:
looks like it's checking if the price is greater than the max price; I think this is probably okay.

Reply:
If you look at the line above that one is checking "price > maxPrice" so I changed this one to check to "price < minPrice"